JBD2: Use the incompat macro for testing the incompat feature.
From: "Aneesh Kumar K.V" <aneesh.kumar@linux.vnet.ibm.com>

JBD2_FEATURE_INCOMPAT_ASYNC_COMMIT need to be checked with JBD2_HAS_INCOMPAT_FEATURE

Signed-off-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
Signed-off-by: Mingming Cao <cmm@us.ibm.com>
---
 fs/jbd2/commit.c   |    2 +-
 fs/jbd2/recovery.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.24/fs/jbd2/commit.c
===================================================================
--- linux-2.6.24.orig/fs/jbd2/commit.c	2008-02-04 11:10:50.000000000 -0800
+++ linux-2.6.24/fs/jbd2/commit.c	2008-02-04 11:12:35.000000000 -0800
@@ -142,7 +142,7 @@ static int journal_submit_commit_record(
 	bh->b_end_io = journal_end_buffer_io_sync;
 
 	if (journal->j_flags & JBD2_BARRIER &&
-		!JBD2_HAS_COMPAT_FEATURE(journal,
+		!JBD2_HAS_INCOMPAT_FEATURE(journal,
 					 JBD2_FEATURE_INCOMPAT_ASYNC_COMMIT)) {
 		set_buffer_ordered(bh);
 		barrier_done = 1;
Index: linux-2.6.24/fs/jbd2/recovery.c
===================================================================
--- linux-2.6.24.orig/fs/jbd2/recovery.c	2008-02-04 10:59:16.000000000 -0800
+++ linux-2.6.24/fs/jbd2/recovery.c	2008-02-04 11:18:25.000000000 -0800
@@ -641,7 +641,7 @@ static int do_one_pass(journal_t *journa
 				if (chksum_err) {
 					info->end_transaction = next_commit_ID;
 
-					if (!JBD2_HAS_COMPAT_FEATURE(journal,
+					if (!JBD2_HAS_INCOMPAT_FEATURE(journal,
 					   JBD2_FEATURE_INCOMPAT_ASYNC_COMMIT)){
 						printk(KERN_ERR
 						       "JBD: Transaction %u "
